core/src/main/resources/error/error-classes.json

```diff
@@ -511,6 +511,12 @@
     ],
     "sqlState" : "38000"
   },
+  "FAILED_PARSE_EMPTY_STRING" : {
+    "message" : [
+      "Failed to parse an empty string for data type <dataType>."
```

Comment:
Please, tell about the SQL config `spark.sql.legacy.json.allowEmptyString.enabled` to users in the error message.

Reply:
I'm conflicted about this. If something is LEGACY, then shouldn't it only apply to existing use cases.
Should we really advertise such configs?
On second thought, I'm not conflicted at all, FWIW I vote to NOT refer to legacy configs.

Reply:
ahh, I missed that it is a legacy config. I agree to don't advertise it.